tests/app/main/views/test_index_page.py

```diff
@@ -120,107 +94,202 @@ def test_check_service_edits_link_is_shown_to_users_with_the_right_role(self, ro
+
+    @staticmethod
+    def _get_mock_framework_response(framework_status):
+        return {"frameworks": [
+            {
+                "id": 1,
+                "name": "Amazing Digital Framework",
+                "slug": "amazing-digital-framework",
+                "status": framework_status,
+            },
+
+        ]}
+
+    @pytest.mark.parametrize('framework_status, header_shown', [
+        ('coming', False),
+        ('open', True),
+        ('pending', True),
+        ('standstill', True),
+        ('live', True),
+        ('expired', False),
+    ])
+    def test_framework_action_lists_not_shown_for_expired_frameworks(
+        self, framework_status, header_shown
+    ):
+        data_api_client.find_frameworks = mock.Mock()
+        data_api_client.find_frameworks.return_value = self._get_mock_framework_response(framework_status)
+
+        self.user_role = "admin-framework-manager"
+        response = self.client.get('/admin')
+        document = html.fromstring(response.get_data(as_text=True))
+
+        assert bool(document.xpath('.//h2[contains(text(),"Amazing Digital Framework")]')) == header_shown
+
+    @pytest.mark.parametrize('framework_status, agreements_shown, stats_shown, comms_shown, contact_shown', [
+        ('coming', False, False, False, False),
+        ('open', False, True, True, True),
+        ('pending', False, True, True, True),
+        ('standstill', True, True, True, True),
+        ('live', True, True, True, True),
+        ('expired', False, False, False, False),
+    ])
+    def test_framework_action_lists_only_contain_actions_for_framework_status(
+            self, framework_status, agreements_shown, stats_shown, comms_shown, contact_shown
+    ):
+        data_api_client.find_frameworks = mock.Mock()
+        data_api_client.find_frameworks.return_value = self._get_mock_framework_response(framework_status)
+        self.user_role = "admin-framework-manager"
+        response = self.client.get('/admin')
+        document = html.fromstring(response.get_data(as_text=True))
+
+        assert bool(document.xpath('.//a[contains(text(),"agreements")]')) == agreements_shown
+        assert bool(document.xpath('.//a[contains(text(),"View application statistics")]')) == stats_shown
+        assert bool(document.xpath('.//a[contains(text(),"Upload communications")]')) == comms_shown
+        assert bool(document.xpath('.//a[contains(text(),"Contact suppliers")]')) == contact_shown
+
+    @pytest.mark.parametrize("role, link_should_be_visible, expected_link_text", [
+        ("admin", False, None),
+        ("admin-ccs-category", True, "View agreements"),
+        ("admin-ccs-sourcing", True, "Countersign agreements"),
+        ("admin-framework-manager", True, "View agreements"),
+        ("admin-manager", False, None),
+    ])
+    def test_framework_action_list_includes_agreements_link(self, role, link_should_be_visible, expected_link_text):
```

Comment:
How does this work without specifying a mock response for `data_api_client.find_frameworks.return_value`?  It feels like it should need a framework before it shows any links?

Reply:
I think it sets framework on base class level using a fixture.